src/SFA.DAS.EAS.Web/Orchestrators/EmployerCommitmentsOrchestrator.cs

```diff
@@ -1075,6 +1099,22 @@ private Task<OrchestratorResponse<ConfirmProviderViewModel>> GetProvider(string
                     }, model.HashedAccountId, externalUser);
         }
 
+        private async Task<(long?, string)> GetTransferConnectionInfo(string hashedAccountId, string transferConnectionCode, string externalUserId)
+        {
+            string transferSenderName = null;
+            long? transferSenderId = null;
+
+            if (!string.IsNullOrEmpty(transferConnectionCode))
+            {
+                var transferConnections = await GetTransferConnectionsNoAuthorizationCheck(hashedAccountId, externalUserId);
+                var transferConnection = transferConnections.TransferConnections.Single(x =>
```

Comment:
This throws an exception if the user has attempted to tamper with values submitted earlier in the multi-stage journey by editing a hidden field. While we don't care about being graceful for such users, we might want to consider throwing a more appropriate exception (argument out of range? invalid operation? I don't know, but something) and a little message in the logs that makes sense for us when we see it. Maybe think about what log level is appropriate (do we want to be warned about this?).

Reply:
Or maybe that's total overkill for something that will likely never happen!

Reply:
I know what you mean. But we do the same with Legal Entities.